build-tools/scripts/PrepareImageDependencies.targets

```diff
@@ -31,6 +31,18 @@
         Condition=" '$(HostOS)' == 'Darwin' And '%(RequiredProgram.Homebrew)' != '' "
         Include="@(RequiredProgram->'%(Homebrew)')"
       />
+      <_BrewTap
+        Condition=" '$(HostOS)' == 'Darwin' And '%(RequiredProgram.Homebrew)' != '' And '%(RequiredProgram.HomebrewTap)' != '' "
+        Include="@(RequiredProgram->'%(HomebrewTap)')"
+      />
+      <_Brew
+        Condition=" '$(HostOS)' == 'Darwin' And '%(RequiredPackage.Homebrew)' != '' "
```

Comment:
Part of the *intention* -- and *maybe* unnecessary? -- was that we'd only emit the `brew install` commands for applications which are not present.

This check can be done for `@(RequiredProgram)`, but not for `@(RequiredPackage)`: how do you check that the package is already installed?  ¯\_(ツ)_/¯

Then again, I'm not sure if this will be an *actual* problem in practice.

I have similar fears about `%(HomebrewTap)`: what happens if the same `brew tap` command is executed multiple times?

In my case, it errors, but it doesn't *error* (?!):

```bash
# I already previously ran `brew tap grendello/xawindeps`

$ brew tap grendello/xawindeps
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
container-diff               fwup                         gradle-completion            kibana                       logstash                     opa                          pgroonga                     postgresql@9.4               pushpin                      unrar                        wtf
file-formula                 git-quick-stats              groonga                      libmagic                     nexus                        osquery                      picat                        profanity                    rocksdb                      weaver

Error: Could not link:
/Users/jon/Dropbox/share/man/man1/brew-cask.1
/Users/jon/Dropbox/share/man/man1/brew.1

Please delete these paths and run `brew update`.
$ echo $?
0
```

I'm not even sure what to make of this state of affairs. :-/

Fortunately, reinstalling a package is a no-op:

```bash
# I previously ran `brew install grendello/xawindeps/mingw-zlib`
$ brew install grendello/xawindeps/mingw-zlib
Warning: grendello/xawindeps/mingw-zlib 1.2.11 is already installed and up-to-date
To reinstall 1.2.11, run `brew reinstall mingw-zlib`
$ echo $?
0
```

Reply:
`homebrew tap` is just a git clone/checkout - nothing bad will happen. As for those errors, note they are warnings, and the error you got in the first case needs `brew doctor` because your homebrew is sick :)